dhis2/dhis2-core
# fix: SMS gateway do not require password and authToken to be send in an update [DHIS2-16695]

### Summary 
Main goal of the PR was to allow updates of a SMS gateway configuration without the need to send in the `password` or `authToken`. If they are blank we keep the existing value, if they have a new value the new value is used. 

In connection with this the attempt was to also make these fields hidden in the API but that requires a wider general change I didn't want to include in this PR. In this context the `/config/sms` endpoint was removed as it is likely unused and outdated and exposes the entire SMS config as one object which makes it even harder to properly protect. 

The refactoring to use lombok `@Getter` and `@Setter` had also revealed bugs in the schema building where getter/setters are not found because of a naming convention mismatch which then causes the `Property` to be broken because of another bug which then causes a stack overflow exception when field filtering the list of gateways. So these issues were addressed in this PR as well. 

### Automatic Testing
Added a tests scenario to ensures the issue with the naming convention is fixed and getter/setters are found even in cases where the convention isn't followed as done by lombok generated methods.

Added test scenarios and asserts to the gateway API tests to make sure an update can be done without sending the fields in question but the fields are still kept internally. 

### Manual Testing 
* create a gateway (of type bulk and also clickatell)
* edit the gateway and leave the passoword/authToken field blank (needs UI update or use of API)
* confirm the fields still have a value after the update (currently the api `/api/systemSettings/keySmsSetting` can be used) otherwise check directly in the DB
